Expensify/App
# Fix forgot password functionality for validated logins

### Details
This PR fixes a deploy blocker and fixes the error mentioned in https://github.com/Expensify/App/pull/5218/files#r712306088. Note this PR does not implement the graceful handling of error messages and resending the validation link, it only fixes the issue preventing non-validated accounts to set a new password.

### Fixed Issues
$ https://github.com/Expensify/App/issues/5384

### Tests
Run the tests from https://github.com/Expensify/App/pull/5218 to confirm no regressions

1. Create an email using [EmailOnDeck](https://www.emailondeck.com).
2. Create and validate the account via `./script/clitools.sh generator:account`, then run `./script/clitools.sh generator:domain` to generate a domain with it as its admin.
3. Create another email using [EmailOnDeck](https://www.emailondeck.com). If you create the account in the same day it should have the same domain.
4. Add this account to the `DEV_EMAILS` const in _.config.local.php.
5. Use this account to sign up for NewDot (don't use any clitools scripts to validate the account). Then, run `vssh php ./Web-Expensify/script/notifyall.php`.
6. You should get an email in EmailOnDeck. Use the link in the email to set a password. 
    - You might have to modify the link format, make sure it's in this format: `<baseURL>/setpassword/<accountID>/<validateCode>`.
7. Populate the password fields and complete the form. You should get logged in.

Confirm already validated logins can continue to newdot after using a new validate link

1. Ask for a reset password link on the sign-in page for an already validated account, click the link in your email
2. Confirm you can set a new password and continue into newdot

### QA Steps
1. Enter email into NewDot sign in screen
2. Select "forgot"
3. Select the link in email
4. Enter new password
5. Continue you can move forward

And regressions on staging:
1. With a fresh email (with no account) in a controlled domain (expensifail should work fine), try to create a new account in newdot.
2. You should get a magic link in your email. Make sure it points to newdot and not oldDot (this is a different bug currently being fixed in another PR). The format should be new.expensify.com/setpassword/<accountID>/<validateCode>. If it isn't, just change it to that format.
3. Populate the password fields and submit the form.
4. You should get logged in.

### Tested On


### Screenshots


#### Web


#### Mobile Web


#### Desktop


#### iOS


#### Android
